deep_qa/models/sentence_selection/siamese_sentence_selector.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any, Dict
+from overrides import overrides
+from keras.layers import Input
+
+from ...data.instances.sentence_selection_instance import SentenceSelectionInstance
+from ...layers.attention.attention import Attention
+from ...layers.wrappers.encoder_wrapper import EncoderWrapper
+from ...training.text_trainer import TextTrainer
+from ...training.models import DeepQaModel
+
+
+class SiameseSentenceSelector(TextTrainer):
+    """
+    This class implements a (generally) Siamese network for the answer
+    sentence selectiont ask. Given a question and a collection of sentences,
+    we aim to identify which sentence has the answer to the question. This
+    model encodes the question and each sentence with (possibly different)
+    encoders, and then does a cosine similarity and normalizes to get a
+    distribution over the set of sentences.
+
+    Note that in some cases, this may not be exactly "Siamese" because the
+    question and sentences encoders can differ.
+    """
+    def __init__(self, params: Dict[str, Any]):
+        self.num_question_words = params.pop('num_question_words', None)
+        self.num_sentence_words = params.pop('num_sentence_words', None)
```

Comment:
This actually won't work.  You're calling this before calling the super class constructor, and the super class will override this with `None`.  You should just remove this line, because the super class handles this parameter.